### PR TITLE
Imbu model type map

### PR DESCRIPTION
**Pull request by BoltzmannBrain**
https://jira.numenta.com/browse/IMBU-115

### CODE REVIEW

**Comment by BoltzmannBrain** [comment]
@oxtopus please review

File: htmresearch/frameworks/nlp/imbu.py [hunk 1]
```diff
@@ -97,6 +97,17 @@ class ImbuModels(object):
   defaultRetina = "en_associative"
   tokenIndexingFactor = 1000
 
+  # Mapping of acceptable model names to names expected by the model factory
+  modelMappings = {
+    "CioWordFingerprint": "CioWordFingerprint",
```
Comment (oxtopus) [inline htmresearch/frameworks/nlp/imbu.py]: Unfortunate duplication here.

Better to do something like:

```
modelMappings = {name:name for name in ClassificationModelTypes.getTypes()}
modelMappings.update("HTM_sensor_knn": "HTMNetwork",
                     "HTM_sensor_simple_tp_knn": "HTMNetwork")
```

File: htmresearch/frameworks/nlp/imbu.py [hunk 2]
```diff
@@ -327,7 +353,7 @@ def main():
                            "header 'Sample'; see readCSV() for more.",
                       required=True)
   parser.add_argument("-m", "--modelName",
-                      choices=list(ClassificationModelTypes.getTypes()),
+                      choices=ImbuModels.modelMappings.keys(),
```
Comment (oxtopus) [inline htmresearch/frameworks/nlp/imbu.py]: Need not use `.keys()`, here.  `choices=ImbuModels.modelMappings` will use the keys by nature of iterating over `dict`s in python.

File: htmresearch/frameworks/nlp/imbu.py (hunk 1 above)
Comment (BoltzmannBrain) [inline htmresearch/frameworks/nlp/imbu.py]: Ah much cleaner, thank you.

**Comment by oxtopus** [comment]
:+1:

**Pull request merged**